dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java

```diff
@@ -447,12 +446,7 @@ private static Set<Version> cloneWorkflowVersion(final Set<Version> originalWork
             detatchedVersion.setName(workflowVersion.getName());
             detatchedVersion.setReference(workflowVersion.getReference());
             SortedSet<SourceFile> sourceFiles = workflowVersion.getSourceFiles();
-            sourceFiles.forEach(sourceFile -> {
-                Gson gson = new Gson();
-                String gsonString = gson.toJson(sourceFile);
-                SourceFile detachedSourceFile = gson.fromJson(gsonString, SourceFile.class);
-                detatchedVersion.addSourceFile(detachedSourceFile);
-            });
+            sourceFiles.forEach(sourceFile -> detatchedVersion.addSourceFile(new SourceFile(sourceFile)));
```

Comment:
I like the copy constructor, it's much clearer what's going on, and a lot more efficient, to boot.  A drawback is that if the constructor gets out of sync and doesn't copy all the fields, it doesn't work right.  Wonder if there is good Java pattern to test that a copy constructor copies all the fields (via reflection?).

Reply:
> A drawback is that if the constructor gets out of sync and doesn't copy all the fields, it doesn't work right. 

This is almost bound to happen at some point too. On the other hand, it looks like version (above) has the same problem, except here in a weird place and with `detatched` spelled wrong. 

I second @svonworl 's idea of having a test that does a sanity check copying at least all the other fields with gson or beanutils

Reply:
Ended up going back to Gson -- found their ExclusionStrategy. Changed this to a static method instead of a copy constructor.